flytekit/annotated/promise.py

```diff
@@ -550,3 +558,128 @@ def __str__(self):
 
     def __repr__(self):
         raise AssertionError(f"Task {self._task_name} returns nothing, NoneType return cannot be used")
+
+
+class NodeOutput(type_models.OutputReference):
+    def __init__(self, node: Node, sdk_type, var: str):
```

Comment:
let us make `sdk_type` `Optional` and default to None?

Reply:
I'm just gonna remove it entirely.